File: FluentFTP.CSharpExamples/GetNameListing.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class GetNameListingExample {

		public static void GetNameListing() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				foreach (var s in conn.GetNameListing()) {
					// load some information about the object
					// returned from the listing...
					var isDirectory = conn.DirectoryExists(s);
					var modify = conn.GetModifiedTime(s);
					var size = isDirectory ? 0 : conn.GetFileSize(s);
				}
			}
		}

		public static async Task GetNameListingAsync() {
			var token = new CancellationToken();
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.ConnectAsync(token);

				foreach (var s in await conn.GetNameListingAsync(token)) {
					// load some information about the object
					// returned from the listing...
					var isDirectory = await conn.DirectoryExistsAsync(s, token);
					var modify = await conn.GetModifiedTimeAsync(s, token);
					var size = isDirectory ? 0 : await conn.GetFileSizeAsync(s, -1, token);
				}
			}
		}

	}
}

File: FluentFTP.VBExamples/GetNameListing.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module GetNameListingExample
		Sub GetNameListing()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()

				For Each s In conn.GetNameListing()
					Dim isDirectory = conn.DirectoryExists(s)
					Dim modify = conn.GetModifiedTime(s)
					Dim size = If(isDirectory, 0, conn.GetFileSize(s))
				Next
			End Using
		End Sub

		Async Function GetNameListingAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.Connect(token)

				For Each s In Await conn.GetNameListing(token)
					Dim isDirectory = Await conn.DirectoryExists(s, token)
					Dim modify = Await conn.GetModifiedTime(s, token)
					Dim size = If(isDirectory, 0, Await conn.GetFileSize(s, -1, token))
				Next
			End Using
		End Function
	End Module
End Namespace
